workshop/content/intermediate/templates/resource-dependencies/_index.md

```diff
@@ -0,0 +1,275 @@
+
+Let’s create a stack, and see the resource creation order in action!
+
+* Make sure you are in the directory: `code/workspace/resource-dependencies`.
+* Open the `resource-dependencies-with-intrinsic-functions.yaml` file.
+* Update the content of the template as you follow along steps on this lab.
+
+Paste the contents of the template snippet shown next in `resource-dependencies-with-intrinsic-functions.yaml` file:
+
+```yaml
+Parameters:
+  SNSTopicName:
+    Description: Enter a name for SNS topic.
+    Type: String
+
+  EmailAddress:
+    Description: Enter an email address to subscribe to SNS topic.
+    Type: String
+
+  SecurityGroupName:
+    Description: Enter a name for security group
+    Type: String
+
+Resources:
+  SNSTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: !Ref SNSTopicName
+
+  SNSTopicSubscription:
+    Type: AWS::SNS::Subscription
+    Properties:
+      Endpoint: !Ref EmailAddress
+      Protocol: email
+      TopicArn: !Ref SNSTopic
+
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security group for CloudFormation lab
+      GroupName: !Ref SecurityGroupName
+
+  SecurityGroupIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !GetAtt SecurityGroup.GroupId
+      IpProtocol: tcp
+      FromPort: 80
+      ToPort: 80
+      CidrIp: 0.0.0.0/0
+```
+
+
+There are four resources in the template snippet you pasted into your template: an [SNS topic](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-topic.html), an [SNS topic subscription](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html), an [EC2 Security Group](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html), and a [SecurityGroupIngress](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-ingress.html). Note the following:
+
+* The logical ID of the SNS topic resource, `SNSTopic`, is referenced with `Ref` in the `TopicArn` property of the `SNSTopicSubscription` resource. This is because the `AWS::SNS::Topic` resource type returns the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) of the SNS topic when you use the `Ref` intrinsic function: for more information, see [SNS topic return values](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-topic.html#aws-properties-sns-topic-return-values). This implies that CloudFormation waits for the creation of `SNSTopic` to complete before initiating the creation of `SNSTopicSubscription`.
```

Comment:
```suggestion
* The logical ID of the topic resource, `SNSTopic`, is referenced with `Ref` in the `TopicArn` property of the `SNSTopicSubscription` resource. This is because the `AWS::SNS::Topic` resource type returns the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) of the topic when you use the `Ref` intrinsic function: for more information, see Amazon SNS topic [return values](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-topic.html#aws-properties-sns-topic-return-values). This implies that CloudFormation waits for the creation of `SNSTopic` to complete before initiating the creation of `SNSTopicSubscription`.
```